numpy/ma/core.py

```diff
@@ -1925,7 +1934,7 @@ def masked_where(condition, a, copy=True):
         cls = MaskedArray
     result = a.view(cls)
     # Assign to *.mask so that structured masks are handled correctly.
-    result.mask = cond
+    result.mask = _shrink_mask(cond)
```

Comment:
So this function is the only real changed behavior, all the other changes are a factorization.

LGTM.